### DISCUSSION TITLE
How is the quary that Linq2DB generates for Firebird supposed to be executed?

### DISCUSSION THREAD

**Discussion by Metadorius** [linq2db/linq2db#4418] [Q&A] [answered]
So Linq2DB generates something like the following for a parametrized select:

```sql
-- SomeConfigName Firebird
DECLARE @SomeParam Integer -- Int32
SET     @SomeParam = 123

SELECT
-- ...
```

This is clearly not part of regular query SQL (tested, fails), hence why I am not sure how can it be executed if I wanted to run the query manually via Rider's SQL console, for example (or IBExpert, or ISQL, or whatever). Maybe that's some basic knowledge but Firebird's docs didn't tell me anything except `EXECUTE BLOCK AS`, prepending which didn't fix all the syntax errors.

**Comment by sdanyliv** [comment]
It is debug information. Historically it shows parameters in SQL Server syntax. 
Under hood `linq2db` uses parameters passed `SqlComand` and do not include declaration section in final SQL.

**Reply by Metadorius** [reply]
Thanks. Is there a way to output something that can be executed as is?

**Reply by sdanyliv** [reply, marked as answer]
Try to turn off parameters:
```cs
db.InlineParameters = true;
```
Or you can turn off them for one query
```cs
query = query.InlineParameters();
```

**Reply by Metadorius** [reply]
I assume this will affect the actual query too, not just the debug display of it?

**Reply by sdanyliv** [reply]
Yes, it will inline parameters as literals.

**Reply by Metadorius** [reply]
I guess there's no other way currently. I'll mark this as answer and submit an issue for this, thanks for the help.

**Reply by sdanyliv** [reply]
Not clear for reproducing something, but i don't know other way how to analyze firebird SQL. Parameters passing is internal ADO.NET technique.
